patroni/validator.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python3
+import logging
+import os
+import socket
+
+from patroni.utils import split_host_port, data_directory_is_empty
+from patroni.ctl import find_executable
+from patroni.dcs import dcs_modules
+from patroni.exceptions import ConfigParseError
+
+logger = logging.getLogger(__name__)
+
+
+def data_directory_empty(data_dir):
+    if os.path.isfile(os.path.join(data_dir, "global", "pg_control")):
+        return False
+    return data_directory_is_empty(data_dir)
+
+
+def validate_connect_address(address):
+    if not isinstance(address, str):
+        raise ConfigParseError("is not a string")
+    else:
+        for s in ["127.0.0.1", "0.0.0.0", "*", "::1"]:
+            if s in address:
+                raise ConfigParseError("must not contain {}".format(s))
+    return True
+
+
+def validate_host_port(host_port, listen=False, connect=False):
+    try:
+        host, port = split_host_port(host_port, None)
+    except (ValueError, TypeError):
+        raise ConfigParseError("contains a wrong value")
+    else:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Comment:
1. `socket.socket()` doesn't support `with` statement in python 2.7
2. do we really need to execute this code if neither listen or connect are set?

Reply:
new commit with fix is on the way :)

Reply:
2. it finds host resolve issues actually.